packages/runner/src/run.ts

```diff
@@ -236,6 +236,8 @@ export async function runTest(test: Test, runner: VitestRunner): Promise<void> {
   await runner.onBeforeRunTask?.(test)
 
   if (test.mode !== 'run' && test.mode !== 'queued') {
+    updateTask('test-prepare', test, runner)
+    updateTask('test-finished', test, runner)
     return
   }
```

Comment:
I remember we discussed this as early as in https://github.com/vitest-dev/vitest/pull/7069#discussion_r1908895272. Happy to see it landing now! This will make resolving test state's on main thread side much easier.